src/test/regress/sql/foreign_key_to_reference_table.sql

```diff
@@ -909,6 +908,55 @@ BEGIN;
   ALTER TABLE test_table_1 ADD COLUMN id int;
 COMMIT;
 DROP TABLE test_table_1, test_table_2, test_table_3;
+-- NOTE: Postgres does not support foreign keys on partitioned tables currently.
+-- However, we can create foreign keys to/from the partitions themselves.
+-- The following tests chech if we create the foreign constraints in partitions properly.
+CREATE TABLE referenced_table(id int PRIMARY KEY, test_column int);
+CREATE TABLE referencing_table(id int, value_1 int) PARTITION BY RANGE (value_1);
+CREATE TABLE referencing_table_0 PARTITION OF referencing_table FOR VALUES FROM (0) TO (2);
+CREATE TABLE referencing_table_2 PARTITION OF referencing_table FOR VALUES FROM (2) TO (4);
+CREATE TABLE referencing_table_4 PARTITION OF referencing_table FOR VALUES FROM (4) TO (6);
+
+-- partitioned tables are not supported as reference tables
+select create_reference_table('referencing_table');
+
+-- partitioned tables are supported as hash distributed table
+SELECT create_reference_table('referenced_table');
+SELECT create_distributed_table('referencing_table', 'id');
+
+-- add foreign constraints in between partitions
+ALTER TABLE referencing_table_0 ADD CONSTRAINT pkey PRIMARY KEY (id);
+ALTER TABLE referencing_table_4 ADD CONSTRAINT fkey FOREIGN KEY (id) REFERENCES referencing_table_0;
+-- add foreign constraint from a partition to reference table
+ALTER TABLE referencing_table_4 ADD CONSTRAINT fkey_to_ref FOREIGN KEY (value_1) REFERENCES referenced_table;
+-- should fail since the data will flow to partitioning_test_4 and it has a foreign constraint to partitioning_test_0 on id column
+INSERT INTO referencing_table VALUES (0, 5);
+-- should succeed on partitioning_test_0
+INSERT INTO referencing_table VALUES (0, 1);
+SELECT * FROM referencing_table;
+-- should fail since partitioning_test_4 has foreign constraint to referenced_table on value_1 column
+INSERT INTO referencing_table VALUES (0, 5);
+INSERT INTO referenced_table VALUES(5,5);
+-- should succeed since both of the foreign constraints are positive
+INSERT INTO referencing_table VALUES (0, 5);
+
+DROP TABLE referenced_table CASCADE;
+DROP TABLE referencing_table;
+
+-- Check if MX works properly
```

Comment:
Well, I said that it's OK to add MX tests here. However, now it seemed unusual to have `start_metadata_sync_to_node ()` (and actually missing `stop_metadata_sync_to_node()`).

Could you simply move these tests to a new file and add it do `mx-schedule`?

Sorry about that :/